[SUREFIRE-2117] Include package, outer classes in XML report for @Nested

andpab opened the pull request:
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [ x ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SUREFIRE) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ x ] Each commit in the pull request should have a meaningful subject line and body.
 - [ x ] Format the pull request title like `[SUREFIRE-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `SUREFIRE-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ x ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ x ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ x ] You have run the integration tests successfully (`mvn -Prun-its clean install`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ x ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

elharo commented:
Run actions/upload-artifact@v3
With the provided path, there will be 8281 files uploaded
Starting artifact upload
For more detailed logs during the artifact upload process, enable step-debugging: https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging
Artifact name is valid!
Error: Artifact path is not valid: /this is: a test_surefire-1617/log.txt. Contains the following character:  Colon :
          
Invalid characters include:  Double quote ", Colon :, Less than <, Greater than >, Vertical bar |, Asterisk *, Question mark ?, Carriage return \r, Line feed \n

andpab commented:
@elharo I fixed the artifact upload in SUREFIRE-2136 and I also fixed the cause of the failed IT under JDK 18 in SUREFIRE-2135. Slawomir just merged both to master, I then rebased this PR to master, so the checks for this PR have all passed now.

slawekjaranowski commented on surefire-its/src/test/java/org/apache/maven/surefire/its/jiras/Surefire2117XmlReportingNestedIT.java:
```diff
@@ -0,0 +1,120 @@
+    public static Iterable<Object[]> artifactVersions()
+    {
+        List<Object[]> args = new ArrayList<>();
+        args.add( new Object[] {"1.0.3", "5.0.3", "1.0.0", "1.0.0"} );
+        args.add( new Object[] {"1.1.1", "5.1.1", "1.0.0", "1.0.0"} );
+        args.add( new Object[] {"1.2.0", "5.2.0", "1.1.0", "1.0.0"} );
+        args.add( new Object[] {"1.3.2", "5.3.2", "1.1.1", "1.0.0"} );
+        args.add( new Object[] {"1.4.2", "5.4.2", "1.1.1", "1.0.0"} );
+        args.add( new Object[] {"1.5.2", "5.5.2", "1.2.0", "1.1.0"} );
+        args.add( new Object[] {"1.6.2", "5.6.2", "1.2.0", "1.1.0"} );
+        args.add( new Object[] {"1.7.1", "5.7.1", "1.2.0", "1.1.0" } );
```
- only `jupiter` param is used in test ... so we can simplify this arrays
- versions 5.8.x, 5.9.x of jupiter are missing in test

slawekjaranowski replied:
ok, I see other tests have the same patterns 😠

andpab replied:
I mindlessly copied this from JUnitPlatformEnginesIT. I'll simplify it and also reduce the number of versions tested. I guess 5.2.0, 5.8.2 and 5.9.1 should be enough for a bugfix-verifying IT like this. I'll also do that for Surefire1914XmlReportingParameterizedTestIT which is another bugfix-verifying IT I added a while ago.

andpab replied:
Pushed the changes to Surefire2117XmlReportingNestedIT. 5.8.2 and 5.9.1 are included now. To reduce the runtime of the test I've dropped all other versions except for 5.2.0 which is the earliest JUnit5 version that seems to be widely used.

Did the same changes to Surefire1914XmlReportingParameterizedTestIT, but in another PR (#588) where I also added 5.8.2 and 5.9.1 to JUnitPlatformEnginesIT and JUnitPlatformIT.

jira-importer commented:
Resolve #2780
